src/Servers/HttpSys/src/RequestProcessing/RequestContext.FeatureCollection.cs

```diff
@@ -728,5 +736,14 @@ public void DelegateRequest(DelegationRule destination)
             Delegate(destination);
             _responseStarted = true;
         }
+
+        void IConnectionLifetimeNotificationFeature.RequestClose()
+        {
+            // Set the connection close feature if the response hasn't sent headers as yet
+            if (!Response.HasStarted)
+            {
+                Response.Headers["Connection"] = "close";
```

Comment:
Are HeaderNames in scope?
```suggestion
                Response.Headers[HeaderNames.Connection] = "close";
```